ui/src/app/certificate_requests/row.tsx

```diff
@@ -0,0 +1,32 @@
+const extractCSR = (csrPemString: string) => {
+    //TODO
+}
+
+const extractCert = (certPemString: string) => {
+    //TODO
+}
+
+type rowProps = {
+    id: number,
+    csr: string,
+    certificate: string
+}
+export default function Row({ id, csr, certificate }: rowProps) {
+    return (
+        <tr>
+            <td className="" data-test-column="id">{id}</td>
+            <td className="u-overflow--visible" data-test-column="details">
+                <p>CN: example.com</p>
+                <p>SAN: example.com, 127.0.0.1, 1.2.3.4.5.56</p>
+            </td>
+            <td className="" data-test-column="status">{certificate == "" ? "outstanding" : (certificate == "rejected" ? "rejected" : "certificate expiry date here")}</td>
+            <td className="" data-test-column="action">
+                <button>Sign</button>
+                <button>Reject</button>
```

Comment:
We can probably do a better job at colors here, every button in this page is black and white.

Reply:
I updated them to all have color, in my opinion the positive and negative colors in the vanilla framework are too bright, and the buttons are very clear at what they do, but if we prefer to have them colored I don't mind.

Reply:
Ok I just checked and I agree that it looks like a Christmas tree. Probably a generic grey Action button would look cleaner. In any case, I recommend looking at MAAS and LXD for inspiration.

Reply:
Although there isn't a button on here, my inspiration was [this page](https://vanillaframework.io/docs/examples/layouts/application/default).

Reply:
This page does looks good.

What do you think of the LXD approach, small buttons with icons that show up when hovering?

![image](https://github.com/canonical/gocert/assets/18486508/af12b15f-4f77-499e-a4c7-1a5565e3cf89)

Reply:
When I tried the icon button, I couldn't find a good icon that could differentiate between rejecting a CSR and deleting a CSR. We could add alt-text that show up under your cursor, but it wouldn't be clear to users.

I'm leaning towards this [example](https://vanillaframework.io/docs/base/tables#responsive), with an action button with a dropdown with clear outcomes.

Reply:
After some conversation, we went with the responsive dropdown